tests/integration/data_sources_and_expectations/expectations/test_expect_column_distinct_values_to_be_in_set.py

```diff
@@ -0,0 +1,107 @@
+from datetime import datetime
+
+import pandas as pd
+import pytest
+
+import great_expectations.expectations as gxe
+from great_expectations.core.result_format import ResultFormat
+from great_expectations.datasource.fluent.interfaces import Batch
+from tests.integration.conftest import parameterize_batch_for_data_sources
+from tests.integration.data_sources_and_expectations.test_canonical_expectations import (
+    ALL_DATA_SOURCES,
+    DATA_SOURCES_THAT_SUPPORT_DATE_COMPARISONS,
+    JUST_PANDAS_DATA_SOURCES,
+)
+
+COL_NAME = "my_col"
+
+ONES_AND_TWOS = pd.DataFrame({COL_NAME: [1, 2, 2, 2]})
+
+
+@parameterize_batch_for_data_sources(data_source_configs=ALL_DATA_SOURCES, data=ONES_AND_TWOS)
+def test_success_complete_results(batch_for_datasource: Batch) -> None:
+    expectation = gxe.ExpectColumnDistinctValuesToBeInSet(column=COL_NAME, value_set=[1, 2])
+    result = batch_for_datasource.validate(expectation, result_format=ResultFormat.COMPLETE)
+    assert result.success
+    assert result.to_json_dict()["result"] == {
+        "details": {
+            "value_counts": [
+                {"value": 1, "count": 1},
+                {"value": 2, "count": 3},
+            ]
+        },
+        "observed_value": [1, 2],
+    }
+
+
+@parameterize_batch_for_data_sources(
+    data_source_configs=ALL_DATA_SOURCES,
+    data=pd.DataFrame({COL_NAME: ["foo", "bar"]}),
+)
+def test_strings(batch_for_datasource: Batch) -> None:
+    expectation = gxe.ExpectColumnDistinctValuesToBeInSet(
+        column=COL_NAME, value_set=["foo", "bar", "baz"]
+    )
+    result = batch_for_datasource.validate(expectation)
+    assert result.success
+
+
+@parameterize_batch_for_data_sources(
+    data_source_configs=DATA_SOURCES_THAT_SUPPORT_DATE_COMPARISONS,
+    data=pd.DataFrame({COL_NAME: [datetime(2024, 11, 19).date(), datetime(2024, 11, 20).date()]}),  # noqa: DTZ001
+)
+def test_dates(batch_for_datasource: Batch) -> None:
+    expectation = gxe.ExpectColumnDistinctValuesToBeInSet(
+        column=COL_NAME,
+        value_set=[datetime(2024, 11, 19).date(), datetime(2024, 11, 20).date()],  # noqa: DTZ001
+    )
+    result = batch_for_datasource.validate(expectation)
+    assert result.success
+
+
+@parameterize_batch_for_data_sources(
+    data_source_configs=JUST_PANDAS_DATA_SOURCES, data=pd.DataFrame({COL_NAME: [1, 2, None]})
+)
+def test_ignores_nulls(batch_for_datasource: Batch) -> None:
+    expectation = gxe.ExpectColumnDistinctValuesToBeInSet(column=COL_NAME, value_set=[1, 2])
+    result = batch_for_datasource.validate(expectation)
+    assert result.success
```

Comment:
very weird that this is our goal behavior

Reply:
Yeah, we're pretty consistent about omitting nulls from the comparisons.